common/server_event_scheduler.cpp

```diff
@@ -0,0 +1,247 @@
+#include "../common/database.h"
+#include "../common/string_util.h"
+#include "server_event_scheduler.h"
+#include "../common/cron/croncpp.h"
+#include <chrono>
+#include <iostream>
+#include <time.h>
+
+ServerEventScheduler::ServerEventScheduler()
+{
+	m_last_polled_minute = -1;
+	m_events             = {};
+	m_active_events      = {};
+}
+
+ServerEventScheduler::~ServerEventScheduler() = default;
+
+void ServerEventScheduler::LoadScheduledEvents()
+{
+	if (!ValidateDatabaseConnection()) {
+		return;
+	}
+
+	std::time_t time = std::time(nullptr);
+	std::tm     *now = std::localtime(&time);
+
+	m_events = ServerScheduledEventsRepository::GetWhere(*m_database, "deleted_at is null");
+	for (auto &e: m_events) {
+
+		auto start = BuildStartTimeFromEvent(e, now);
+		auto end   = BuildEndTimeFromEvent(e, now);
+
+		// data excluded from output because it can be very large
+
+		LogScheduler(
+			"Loaded Event ({}) [{}] type [{}] start [{}/{}/{} {}:{}:00] end [{}/{}/{} {}:{}:00] cron [{}] created [{}]",
```

Comment:
Specifying a 0 padded minimum width `{:02}` for the hours:minutes might make this easier to read in the output
`start [2/3/2021 15:1:00] end [2/15/2021 19:7:00]`
would become
`start [2/3/2021 15:01:00] end [2/15/2021 19:07:00]`

Reply:
This was a great suggstion, done